sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/KeyAsyncClient.java

```diff
@@ -97,11 +98,36 @@
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<Key>> createKey(String name, KeyType keyType) {
+        return withContext(context -> createKey(name, keyType, context));
+    }
+
+    /**
+     * Creates a new key and stores it in the key vault. The create key operation can be used to create any key type in
+     * key vault. If the named key already exists, Azure Key Vault creates a new version of the key. It requires the {@code keys/create} permission.
+     *
+     * <p>The {@link KeyType keyType} indicates the type of key to create. Possible values include: {@link KeyType#EC EC},
+     * {@link KeyType#EC_HSM EC-HSM}, {@link KeyType#RSA RSA}, {@link KeyType#RSA_HSM RSA-HSM} and {@link KeyType#OCT OCT}.</p>
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Creates a new EC key. Subscribes to the call asynchronously and prints out the newly created key details when a response has been received.</p>
+     * <pre>
+     * keyAsyncClient.createKey("keyName", KeyType.EC, context).subscribe(keyResponse -&gt;
+     *   System.out.printf("Key is created with name %s and id %s \n", keyResponse.value().name(), keyResponse.value().id()));
+     * </pre>
+     *
+     * @param name The name of the key being created.
+     * @param keyType The type of key to create. For valid values, see {@link KeyType KeyType}.
+     * @param context Additional context that is passed through the Http pipeline during the service call.
+     * @throws ResourceModifiedException if {@code name} or {@code keyType} is null.
+     * @throws HttpRequestException if {@code name} is empty string.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#value() value} contains the {@link Key created key}.
+     */
+    Mono<Response<Key>> createKey(String name, KeyType keyType, Context context) {
```

Comment:
No need for such prolific JavaDoc on a non-public API. I would likely remove all of this.